docs/NFF-0.1-spec.norg

```diff
@@ -0,0 +1,875 @@
+    limitations because insertion of (for example) code blocks is not possible. To alleviate this
+    deficiency, the {# indent segment extension} exists, which increases the indentation of the
+    tree by one level allowing nesting of more complex elements to occur.
+    The syntax is the following:
+    - The indent segment extension starts with a single `\` char
+    - This must be _immediately_ followed by a {# line ending}
+    -- Thus, the start node of this extension can be seen as a {# trailing modifier}
+    - The contents follow
+    - The extension ends when:
+    -- Another {# nestable detached modifier} occurs
+    -- Or when the indentation is manually closed with one of the {# delimiting modifiers}
+     +TODO(vhyrro): discuss what truly "automatically" ends the indent segment. I propose that the only way
+     to auto-close an indent segment is to use a structural detached modifier or a modifier of the same
+     type as the parent of the indent segment; i.e. if you attach an indent segment to a level 1
+     unordered list item then creating another level 1 unordered list item auto-closes the indent
+     segment. However, creating a level 1 ordered list segment inside would be okay.+
+
+    @code norg
+    - \
+      The indent segment extension has started.
+
+      I can now include multiple paragraphs within this list item.
+
+      @code lua
+      print("Even code blocks are allowed!")
+      @end
+
+      ^ Range-able detached modifiers can be inserted, too.
+
+      ---
+    - Another nestable detached modifier ends the indent segment
+    - However, in this example I already closed the indent segment manually with `---`
+    @end
+
+* Tags
+  Tags are _Norg_'s official way of extending its own syntax.
+
+  There are 3 different tag types, each with their own way of changing the way text in _Norg_ is
+  interpreted. Before we discuss those, however, we should discuss the syntax rules for tags:
+  - A tag is similar to a {# detached modifiers}[detached modifier] in the sense that it must begin
+    at the beginning of a line with optional whitespace (but nothing else) preceding it.
+  - After that you will encounter the special tag character (`|`, `@` or `#`), /none/ of which are
+    attached modifiers - if those chars were attached modifiers then we would have ambiguity on our
+    hands. The special tag character is then /immediately/ followed by text, which becomes the /tag
+    name/. Said tag name can consist of any {# alphanumeric characters}[alphanumeric character]
+    and/or `-`, `_` and `.`.
+  - Tags can have their names delimited by a `.` in order to create a "hierarchy", e.g.
+    `document.meta`.
+  - \
+    After a whitespace character any number of parameters on the same line may follow:
+    @code norg
+    |tag-name.subtag parameter1 parameter2
+    @end
+
+    Parameters may consist of any character (apart from a {# line break}, of course).
```

Comment:
We need to discuss multi-word parameters here and the fact that this disallows the use of double quotes `"` in parameters